client/src/main/java/io/pravega/client/stream/impl/ControllerResolverFactory.java

```diff
@@ -137,19 +141,12 @@ public void start(Listener listener) {
                         .loadBalancerFactory(RoundRobinLoadBalancerFactory.getInstance())
                         .usePlaintext(true)
                         .build());
-                this.scheduledExecutor = Executors.newScheduledThreadPool(1,
-                        new ThreadFactoryBuilder().setNameFormat("fetch-controllers-%d").setDaemon(true).build());
-                this.scheduledExecutor.scheduleWithFixedDelay(
-                        this::getControllers, 0L, 120L, TimeUnit.SECONDS);
-            } else {
-                // Use the bootstrapped server list as the final set of controllers.
-                final ResolvedServerInfoGroup serverInfoGroup = ResolvedServerInfoGroup.builder().addAll(
-                        this.bootstrapServers.stream().map(ResolvedServerInfo::new)
-                                .collect(Collectors.toList()))
-                        .build();
-                log.info("Updating client with controllers: {}", serverInfoGroup);
-                this.resolverUpdater.onUpdate(Collections.singletonList(serverInfoGroup), Attributes.EMPTY);
             }
+
+            // This should be a single threaded executor to ensure invocations of getControllers are serialized.
+            this.scheduledExecutor = Executors.newScheduledThreadPool(1,
```

Comment:
I have not addressed this - PR #1543. As this will make every application of Controller client pass an executor. Since we only have limited use of this executor internally, am not sure if we really want to expose this outside. Please let me know if you think this is really necessary, will address it in PR #1543